src/window/mod.rs

```diff
@@ -235,69 +234,68 @@ pub fn main_loop(
     event_loop: EventLoop<UserEvent>,
 ) -> Result<(), EventLoopError> {
     let cmd_line_settings = SETTINGS.get::<CmdLineSettings>();
-    let (txtemp, rx) = channel::<Event<UserEvent>>();
-    let mut tx = Some(txtemp);
-    let mut render_thread_handle = Some(thread::spawn(move || {
+    let (tx, rx) = channel::<Event<UserEvent>>();
+
+    let render_thread_handle = thread::spawn(move || {
         let mut window_wrapper = WinitWindowWrapper::new(window, initial_window_size);
         let mut update_loop = UpdateLoop::new(
             cmd_line_settings.vsync,
             cmd_line_settings.idle,
             &window_wrapper.windowed_context,
         );
-        #[allow(unused_assignments)]
+
         loop {
+            if !RUNNING_TRACKER.is_running() {
+                break;
+            }
+
             let (wait_duration, _) = update_loop.get_event_wait_time();
-            let event = if wait_duration.as_nanos() == 0 {
-                rx.try_recv()
-                    .map_err(|e| matches!(e, TryRecvError::Disconnected))
-            } else {
-                rx.recv_timeout(wait_duration)
-                    .map_err(|e| matches!(e, RecvTimeoutError::Disconnected))
-            };
+            let event = rx
+                .recv_timeout(wait_duration)
```

Comment:
I seem to recall that I had some problems with a zero duration and `recv_timeout` when I wrote the code originally. But I can't find anything in the documentation or in the rust bug reports. I also can't reproduce it, so maybe either I had a different code structure when I noticed the problem, or my code had some other problem at that time.

Reply:
I also looked around for reasons not to call it with a zero duration, just in case, but didn't see anything. I traced through the `recv_timeout` source code as well to make sure, and didn't see anything that would lead me to think that anything bad would happen. Everything appears to be running fine this way.